### PR TITLE
.time functionality

### PR DESCRIPTION
**Pull request by Kyhz**

### CODE REVIEW

File: commands/time.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* Usage: .time [zone]
+ * Ex: .time Asia/Beijing, .time JST
+ */
+
+import {Command} from './command.js';
+var tz = require('timezone/loaded');
```
Comment (IanMitchell) [inline commands/time.js]: change this line to `import timezone from 'timezone/loaded';`, and all `tz` references to `timezone` for me?

File: commands/time.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export class Time extends Command {
+  message(from, to, text, message) {
+    let timezoneMatch = text.match(/^[.!]time (.*)/);
```
Comment (IanMitchell) [inline commands/time.js]: Can we wrap this in a promise? See https://github.com/IanMitchell/ARX-7/pull/18/files#diff-1e17032847c481c35012066d0160b4ecR8 for an example

File: commands/time.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+  }
+
+  getTime(timezone, callback) {
+    let formattedTime = tz(new Date(), "%c", timezone);
+    callback(formattedTime);
+  }
+
+  // Timezone abbreviations are ambiguous but I'm smarter than timezones
+  abbreviationMap(abbr) {
```
Comment (IanMitchell) [inline commands/time.js]: Could we do this with the new Map data structure? http://es6-features.org/#MapDataStructure

**Comment by Kyhz** [comment]
Addressed comments. First time using ES6 on anything so I appreciate the tips on new features.

**Comment by IanMitchell** [comment]
ES6 is great :smile:

File: arx7.js [hunk 4]
```diff
@@ -23,7 +24,8 @@ let commands = [
   new Imgur(client),
   new Order(client),
   new Twitter(client),
-  new Youtube(client)
+  new Youtube(client),
+  new Time(client)
```
Comment (IanMitchell) [inline arx7.js]: Bit of a nit - can we insert this above Twitter? As the list of plugins grows, keeping them alphabetical would be nice

File: commands/time.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * Ex: .time Asia/Beijing, .time JST
+ */
+
+import {Command} from './command.js';
+import timezone from 'timezone/loaded';
+
+export class Time extends Command {
+  message(from, to, text, message) {
+    let timezoneMatch = text.match(/^[.!]time (.*)/);
+    return new Promise((resolve, reject) => {
```
Comment (IanMitchell) [inline commands/time.js]: Oh, wrapping the if is interesting... didn't even think of that. I'm gonna adopt that model, it'll clean my code up a bit!

**Comment by IanMitchell** [comment]
Just merged in my changes, which means this branch is gonna have some conflicts now - mind updating it accordingly? I can do it too if you're not sure what needs to change

File: src/arx7.js [hunk 6]
```diff
@@ -1,11 +1,12 @@
 import debug from 'debug';
 import irc from 'irc';
-import config from '../config';
+import config from '../config.json';
```
Comment (IanMitchell) [inline src/arx7.js]: With Node, you don't need to specify the `.json` - it works automatically :smile:

**Pull request merged**